### PR TITLE
Fix#6929: docker-compose services should wait until OpenMetadata DB is created

### PR DESCRIPTION
**Pull request by nahuelverdugo**
Fixes: #6929 
Based on PR: #6598  

### Describe your changes :
- Added health checks in `docker-compose` files to the following services:
  - `mysql`
  - `postgresql`
  - `openmetadata-server`
- Removed unnecessary waits for services inside `sh` scripts used in `Dockerfile` files
- Aligned names for services and containers

### Type of change :
- [x] Bug fix
- [x] Improvement

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
@akash-jain-10 @Vj-L @ulixius9 @vivekratnavel @harshach

### CODE REVIEW

File: docker/local-metadata/docker-compose-dev-postgres.yml [hunk 1]
```diff
@@ -11,18 +11,11 @@
 
 version: "3.9"
 services:
-  postgres:
-    build:
-      context: ../../.
-      dockerfile: docker/local-metadata/Dockerfile_postgres
-    container_name: openmetadata_postgres
```
Comment (pmbrull) [inline docker/local-metadata/docker-compose-dev-postgres.yml:18]: Why are we not using our image? In there we should be having the latest postgres already but with the users ready

**Review by pmbrull — APPROVED** [review]
Great changes, thanks for the PR. Left a qq if you have the time. Thanks!

File: docker/local-metadata/docker-compose-dev-postgres.yml (hunk 1 above)
Comment (nahuelverdugo) [reply docker/local-metadata/docker-compose-dev-postgres.yml:18]: I have aligned this file with what we have in `docker-compose-dev.yml`. I can roll back this change and align it with `docker-compose-dev.yml`. What should I do?

Comment (pmbrull) [reply docker/local-metadata/docker-compose-dev-postgres.yml:18]: LGTM then, thanks for raising

Comment (nahuelverdugo) [reply docker/local-metadata/docker-compose-dev-postgres.yml:18]: We have agreed offline to leave this change.

File: docker/airflow/Dockerfile [hunk 2]
```diff
@@ -4,10 +4,9 @@ RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add -
 RUN curl https://packages.microsoft.com/config/debian/11/prod.list > /etc/apt/sources.list.d/mssql-release.list
 RUN apt-get update \
     && apt-get install -y \
-        gcc libsasl2-dev gnupg build-essential libssl-dev libffi-dev \
-        librdkafka-dev unixodbc-dev libxml2 libevent-dev --no-install-recommends  \
+        build-essential gcc gnupg libevent-dev libffi-dev libpq-dev librdkafka-dev libsasl2-dev libsasl2-modules  \
```
Comment (akash-jain-10) [inline docker/airflow/Dockerfile:7]: looks like we are missing on `openjdk-11-jre` that is required for deltalake connector! Let's add that in here!

File: docker/local-metadata/docker-compose-dev-postgres.yml [hunk 3]
```diff
@@ -11,18 +11,11 @@
...
+  postgresql:
+    image: openmetadata/postgresql:latest
```
Comment (akash-jain-10) [inline docker/local-metadata/docker-compose-dev-postgres.yml:15]: Let's instead build the postgresql using `build`, `context` and `dockerfile` for local development.

File: docker/metadata/openmetadata-start.sh [hunk 4]
```diff
@@ -10,10 +10,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-MYSQL="${MYSQL_HOST:-mysql}":"${MYSQL_PORT:-3306}"
-while ! nc -z -w 5 "${MYSQL}";
-  do echo "Trying to connect to ${MYSQL}"; sleep 5;
-done
+echo "Starting OpenMetadata Server";
```
Comment (akash-jain-10) [inline docker/metadata/openmetadata-start.sh:13]: I will need to test out this change with OpenMetadata Helm Charts, creating a local docker image! Let me get back here testing post that!

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: docker/local-metadata/docker-compose-dev-postgres.yml (hunk 3 above)
Comment (nahuelverdugo) [reply docker/local-metadata/docker-compose-dev-postgres.yml:15]: Should I update the same for `docker/local-metadata/docker-compose-dev.yml`?

File: docker/airflow/Dockerfile (hunk 2 above)
Comment (nahuelverdugo) [reply docker/airflow/Dockerfile:7]: Should I add also the same dependencies that we have in `ingestion/Dockerfile`?

Comment (akash-jain-10) [reply docker/airflow/Dockerfile:7]: we can ignore `python3.9-dev`, `netcat` and `wget`

File: docker/local-metadata/docker-compose-dev-postgres.yml (hunk 3 above)
Comment (akash-jain-10) [reply docker/local-metadata/docker-compose-dev-postgres.yml:15]: Looks like it's already update there!

File: docker/metadata/openmetadata-start.sh (hunk 4 above)
Comment (nahuelverdugo) [reply docker/metadata/openmetadata-start.sh:13]: Feel free to put it back if OpenMetadata Helm Charts test fails after this change.

Comment (akash-jain-10) [reply docker/metadata/openmetadata-start.sh:13]: Tested out this change! Looks good on Kubernetes as well! Let's roll in these changes! :)

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**